R/process_observers.R

```diff
@@ -0,0 +1,170 @@
+      new_primaryobs <- readline(":")
+      
+      if(new_primaryobs == "QUIT") {save_observer_table(observer_table); #save and quit
+        return("Function Ended")}
+      if(new_primaryobs == "!QUIT") {return("Function Ended")} #quit without saving
+      if(new_primaryobs == "NA") {}#do nothing
+      else {
+        #add new row to overall observer_table
+        observer_table <- observer_table %>% 
+          add_row(county = selected_county,
+                  route = rocombos$route_num[i], 
+                  observers = rocombos$observers[i],
+                  primary_observer = new_primaryobs)
+        #update the county_observer_table so the new info shows up if more than one new observer is going to be added to the route
+        county_observer_table <- observer_table %>% filter(county == selected_county)
+      } #end else statement about adding a new primary observer or not
+      
+      #give user the option to edit a row that already exists in that route
+      # edit_row <- readline("Do you want to edit an existing row for this route?
+      #                       Type Y to edit,
+      #                       Type anything else to move to next new route/obersever combo")
+      # if(edit_row == "Y") {
+      #   edit_row <- readline("Type the full observers column to be changed:")
+      #   #check that observers column exists, quit if it doesn't
+      #   if(county_observer_table %>% filter(route == rocombos$route_num[i]) %>% 
+      #                                filter(observers == edit_row) %>% nrow() > 0) {
+      #   new_primaryobs <- readline("What should the new primary observer be?:")
+      #   #mutate that row
+      #   observer_table <- observer_table %>% mutate(
+      #                     primary_observer = replace(primary_observer, 
+      #                                                county == selected_county & #match county
+      #                                                route == rocombos$route_num[i] & #& route
+      #                                                observers == edit_row, #& observers
+      #                                                new_primaryobs)) #replace w/new primaryobs
+      #   } else {print("Cannot find observers as typed, moving to next new route/observer combo" )}
+      
+    } #end else statement about this observer/route combo not being on the conversion table
+  }  #end for loop (done going through all the rocombos)
+  
+  #save updated version of observer conversion table
+  save_observer_table(observer_table)
+  print("No more new route/observer combos. Done!")
+} #end function
+
+
+
+#' Pull out observers from the checklist_comments column that process_comments missed
+#' @param mbbs_county mbbs data.frame
+#' @importFrom stringr str_replace_all str_detect
+#' @importFrom dplyr mutate case_when
+#' @export
+observers_extractor <- function(mbbs_county) {
+  
+  #fix unicode 
+  mbbs_county$checklist_comments <- mbbs_county$checklist_comments %>%
```

Comment:
Given the [changes here](https://github.com/nc-minibbs/mbbs/pull/36/files#diff-6b772abb1f5d57c91f4f9b022ae3cbd82c1a71e6cc9361df40a5513c41e80246R11) is this step still necessary?